### PR TITLE
fix(components): show no data indicator on mutation over time instead of error

### PR DESCRIPTION
**Pull request by JonasKellerer**
Resolves: #501
Resolves: #487

### Summary
Shows a "No data available" indication, when there is no data for the lapis filters. Also adds a no data message, for the filters the user can select in the component itself.

### Screenshot

When the user selects a too narrow filter in the component
grafik

### PR Checklist

~~- [ ] All necessary documentation has been adapted.~~
- [x] The implemented feature is covered by an appropriate test.

### CODE REVIEW

**Review by fengelniederhammer — CHANGES_REQUESTED** [review]
I'm still seeing the error and 
```
TypeError: allDates[0] is undefined
    queryOverallMutationData queryMutationsOverTime.ts:71
``` 
when I enter a `dateFrom` in 2025.

Also something seems to be weird with the webcomponents Storybook. When I change the filters there, the component doesn't really update?

**Comment by JonasKellerer** [comment]
> Also something seems to be weird with the webcomponents Storybook. When I change the filters there, the component doesn't really update?

Resolved by #510

File: components/src/preact/mutationsOverTime/mutations-over-time.stories.tsx [hunk 1]
```diff
@@ -85,3 +85,73 @@ export const ShowsMessageWhenTooManyMutations: StoryObj<MutationsOverTimeProps>
...
+        lapisDateField: 'date',
+    },
+    play: async ({ canvas }) => {
+        await waitFor(() => expect(canvas.getByText('No data available.', { exact: false })).toBeVisible(), {
+            timeout: 10000,
+        });
+    },
+};
+
+export const ShowsNoDataMessageWhenThereAreNoDatesInFilter: StoryObj<MutationsOverTimeProps> = {
```
Comment (fengelniederhammer) [inline components/src/preact/mutationsOverTime/mutations-over-time.stories.tsx:107]: This story still calls the real LAPIS. We should mock the data to get properly stable tests.

**Review by fengelniederhammer — APPROVED** [review]
LGTM :+1:

File: components/src/preact/mutationsOverTime/mutations-over-time.stories.tsx (hunk 1 above)
Comment (JonasKellerer) [reply components/src/preact/mutationsOverTime/mutations-over-time.stories.tsx:107]: Good catch. I added the corresponding mock.

**Pull request merged**